conda_env/specs/yaml_file.py

```diff
@@ -0,0 +1,19 @@
+import os
+
+from .. import env
+
+
+class YamlFileSpec(object):
+    def __init__(self, filename=None, name=None, **kwargs):
+        self.filename = filename
+        self.name = name
+
+    def can_handle(self):
+        self._environment = env.from_file(self.filename)
```

Comment:
I have an idea about this.
Normally a `detect` method should raise an exception when failing to detect. Following that behavior we can raise an exception that has logged why every spec replied `False` to `can_handle`. And we can show that error to the user. Something like:
```
Couldn't find an environment.yml file.
Couldn't find malev/project in Binstar.org
...
```